src/components/datagrid/controls/data_grid_toolbar.tsx

```diff
@@ -48,11 +48,18 @@ export const EuiDataGridToolbar = ({
     ? true
     : gridWidth > minSizeForControls || isFullScreen;
 
-  // need to safely access those Web APIs
-  if (typeof document !== 'undefined') {
-    // When data grid is full screen, we add a class to the body to remove the extra scrollbar
-    document.body.classList.toggle(GRID_IS_FULLSCREEN_CLASSNAME, isFullScreen);
-  }
+  useEffect(() => {
+    // When data grid is full screen, we add a class to the body to remove the extra scrollbar and stay above any fixed headers
+    if (isFullScreen && typeof document !== 'undefined') {
+      document.body.classList.add(GRID_IS_FULLSCREEN_CLASSNAME);
+
+      return () => {
+        if (typeof document !== 'undefined') {
+          document.body.classList.remove(GRID_IS_FULLSCREEN_CLASSNAME);
+        }
+      };
+    }
+  }, [isFullScreen]);
```

Comment:
Super awesome fix, thanks! Do you think this needs a changelog entry or probably not?

Reply:
Wouldn't hurt to give it a callout

Reply:
And I think that actually closes this: https://github.com/elastic/eui/issues/5092

Reply:
Nice, I added a changelog entry and a closes commit (which I think should auto close on merge to main?)

Reply:
I honestly don't know if a commit message will. But might as well add it to your PR summary as well.